scripts/irods/test/test_experimental_parallel_remove.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import print_function
+import copy
+import logging
+import sys
+if sys.version_info >= (2, 7):
+    import unittest
+else:
+    import unittest2 as unittest
+import os
+import pprint
+import time
+import json
+import tempfile
+import shutil
+
+from .. import test
+from . import settings
+from .. import lib
+from ..configuration import IrodsConfig
+from ..controller import IrodsController
+from .. import paths
+from . import resource_suite
+from . import session
+
+class TestParallelRecursiveRemove(session.make_sessions_mixin([], []), unittest.TestCase):
+
+    def setUp(self):
+        super(TestParallelRecursiveRemove, self).setUp()
+
+    def tearDown(self):
+        super(TestParallelRecursiveRemove, self).tearDown()
+
+    def test_remove_object(self):
+        with session.make_session_for_existing_admin() as admin:
+            try:
+                file_name = 'test_remove_object'
+                lib.make_file(file_name, 1024)
+
+                admin.assert_icommand(['iput', '-f', file_name])
+                admin.assert_icommand(['ils', '-l'], 'STDOUT_SINGLELINE', file_name)
+
+                logical_path = '/tempZone/home/rods/' + file_name
+
+                endpoint = 'recursive_remove'
+                options  = """{"logical_path" : "/tempZone/home/rods/test_remove_object"}"""
+                out,_ = lib.execute_command(['irods_api_test_harness', options, endpoint])
+
+                admin.assert_icommand(['ils', '-l', '/tempZone/trash/home/rods/'], 'STDOUT_SINGLELINE', file_name)
+            finally:
+                os.remove(file_name)
+
+    def test_remove_object_no_trash(self):
+        with session.make_session_for_existing_admin() as admin:
+            try:
+                admin.assert_icommand(['irmtrash'])
+
+                file_name = 'test_remove_object'
+                lib.make_file(file_name, 1024)
+
+                admin.assert_icommand(['iput', '-f', file_name])
+                admin.assert_icommand(['ils', '-l'], 'STDOUT_SINGLELINE', file_name)
+
+                logical_path = '/tempZone/home/rods/' + file_name
+
+                endpoint = 'recursive_remove'
+                options  = """{"logical_path" : "/tempZone/home/rods/test_remove_object",
+                               "no_trash"     : true}"""
+                out,_ = lib.execute_command(['irods_api_test_harness', options, endpoint])
+
+                admin.assert_icommand(['ils', '-l', '/tempZone/trash/home/rods/test_remove_object'], 'STDERR_SINGLELINE', 'ERROR')
```

Comment:
what error specifically?

Reply:
remote addresses: 152.54.8.141 ERROR: lsUtil: srcPath /tempZone/home/rods/nopes does not exist or user lacks access permission